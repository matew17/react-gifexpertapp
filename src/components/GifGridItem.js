import React from 'react'
import PropTypes from 'prop-types'
import './../index.css';
function GifGridItem({ url, title }) {
    return (
        <div className="card animate__animated animate__bounce">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}

export default GifGridItem
