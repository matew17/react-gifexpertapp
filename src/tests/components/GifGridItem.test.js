import { shallow } from "enzyme";
import React from "react";

import GifGridItem from './../../components/GifGridItem';

describe('GifGridItem.test', () => {
    const title = 'titulo';
    const url = 'https://localhost.com/image.jpg';
    const wrapper = shallow(
        <GifGridItem
            title={title}
            url={url}
        />);

    test('Debe mostrar el componente correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar un parrrafo con el titulo ', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    });

    test('Debe tener la imagen igual al url y title de los props ', () => {
        const img = wrapper.find('img');

        expect(img.prop('alt')).toBe(title);
        expect(img.prop('src')).toBe(url);
    });

    test('Debe tener DIV animate__bounce ', () => {
        const div = wrapper.find('div');

        expect(div.hasClass('animate__bounce')).toBe(true);
    });
});
