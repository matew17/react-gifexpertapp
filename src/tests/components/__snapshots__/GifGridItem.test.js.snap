// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem.test Debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="titulo"
    src="https://localhost.com/image.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
